### ISSUE TITLE
A variable resolves to return value in variable type

### ISSUE THREAD

**Issue by deadok22** [go-lang-plugin-org/go-lang-idea-plugin#2502]
In the example below ```rv``` resolves to ```rv``` in ```bar```'s return value. It also leads to a false-positive "No new variables on left side of :=" warning.
```go
func foo() int {
	var bar func() (rv int)
	bar = func() int { return 11 }
	rv := bar()
	return rv
}
```

Plugin version: 0.11.1316

**Comment by zolotov** [comment]
I believe I fixed this in 0.11.1323, please try the new build

**Comment by dlsniper** [comment]
I can confirm it's fixed in 0.11.1323

**Comment by deadok22** [comment]
It's all good in 0.11.1343.
